TransactionProcessor.BusinessLogic/EventHandling/TransactionDomainEventHandler.cs

```diff
@@ -179,6 +191,62 @@ private async Task<TokenResponse> GetToken(CancellationToken cancellationToken)
             }
         }
 
+        /// <summary>
+        /// Handles the specific domain event.
+        /// </summary>
+        /// <param name="domainEvent">The domain event.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        private async Task HandleSpecificDomainEvent(CustomerEmailReceiptRequestedEvent domainEvent,
+                                                     CancellationToken cancellationToken)
+        {
+            this.TokenResponse = await this.GetToken(cancellationToken);
+
+            TransactionAggregate transactionAggregate = await this.TransactionAggregateManager.GetAggregate(domainEvent.EstateId, domainEvent.TransactionId, cancellationToken);
+
+            var merchant = await this.EstateClient.GetMerchant(this.TokenResponse.AccessToken, domainEvent.EstateId, domainEvent.MerchantId, cancellationToken);
+
+            // Determine the body of the email
+            String receiptMessage = await this.TransactionReceiptBuilder.GetEmailReceiptMessage(transactionAggregate.GetTransaction(), merchant, cancellationToken);
+
+            // Send the message
+            await this.SendEmailMessage(this.TokenResponse.AccessToken, domainEvent.EstateId, "Transaction Successful", receiptMessage, domainEvent.CustomerEmailAddress, cancellationToken);
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/stuart_ferguson1/TransactionProcessor/pullRequest?prid=6301818)